engines/rancher_on_eks/app/models/rancher_on_eks/fqdn.rb

```diff
@@ -14,5 +14,9 @@ def self.load
     def save!
       KeyValue.set(:fqdn, @value)
     end
+
+    def dns_available?
+      system("ping -c 1 -W 1 #{@value}")
```

Comment:
Not sure whether that's the best way to test whether DNS is available. The ping may fail due to other reasons than `@value` not being resolvable. The system may not be reachable, there may be packet loss, or ICMP traffic may be filtered. Wouldn't it make more sense to simply try to resolve the fqdn?

Reply:
Yes, I did not want to add a new Gem
it seems the library that does that is in the stdlib

Reply:
I would have the same comment/question. Is Ruby similar to Python in terms of "ask for forgiveness"? Using a try/except style handling instead of lbyl?

Reply:
So couldn't the system('ping ...') call be replaced by something like Addrinfo.getaddrinfo()? That's in the stdlib AFAICS.

Reply:
@jgleissner I updated the code to the validations we had planned (valid URL and hosted zone for subdomain)